examples/openai_chat_completion_client_with_tools.py

```diff
@@ -0,0 +1,143 @@
+from openai import OpenAI
```

Comment:
Can do :)

Reply:
I was hesitant to add tests that pull each of the models supported by tool parsers and use them to generate calls because I know that's time+memory+compute+storage expensive - just confirming that's ok